.travis.yml

```diff
@@ -2,53 +2,86 @@ language: generic
 
 matrix:
   include:
+    - os: osx
+      env:
+        - ARCH=x86_64
+        - MINIFORGE_NAME="Miniforge-pypy3"
+        - OS_NAME="Linux"
+
+    - os: osx
+      env:
+        - ARCH=x86_64
+        - MINIFORGE_NAME="Miniforge3"
+        - OS_NAME="Linux"
+
     - os: linux
       env:
         - ARCH=aarch64
         - DOCKER_ARCH=arm64v8
         - DOCKERIMAGE=condaforge/linux-anvil-aarch64
         - MINIFORGE_NAME="Miniforge3"
+        - OS_NAME="Linux"
 
     - os: linux
       env:
         - ARCH=x86_64
         - DOCKER_ARCH=amd64
         - DOCKERIMAGE=condaforge/linux-anvil-comp7
         - MINIFORGE_NAME="Miniforge3"
+        - OS_NAME="Linux"
 
     - os: linux
       env:
         - ARCH=ppc64le
         - DOCKER_ARCH=ppc64le
         - DOCKERIMAGE=condaforge/linux-anvil-ppc64le
         - MINIFORGE_NAME="Miniforge3"
+        - OS_NAME="Linux"
 
     - os: linux
       env:
         - ARCH=aarch64
         - DOCKER_ARCH=arm64v8
         - DOCKERIMAGE=condaforge/linux-anvil-aarch64
         - MINIFORGE_NAME="Miniforge-pypy3"
+        - OS_NAME="Linux"
 
     - os: linux
       env:
         - ARCH=x86_64
         - DOCKER_ARCH=amd64
         - DOCKERIMAGE=condaforge/linux-anvil-comp7
         - MINIFORGE_NAME="Miniforge-pypy3"
+        - OS_NAME="Linux"
 
     - os: linux
       env:
         - ARCH=ppc64le
         - DOCKER_ARCH=ppc64le
         - DOCKERIMAGE=condaforge/linux-anvil-ppc64le
         - MINIFORGE_NAME="Miniforge-pypy3"
+        - OS_NAME="Linux"
 
 script:
   - if [[ "$TRAVIS_TAG" != "" ]]; then export MINIFORGE_VERSION=$TRAVIS_TAG; fi
-  - bash build_miniforge.sh
+  - if [[ "$OS_NAME" == "Linux" ]]; then bash build_miniforge.sh; fi
+  - if [[ "$OS_NAME" == "OSX" ]]; then
+        echo "Installing a fresh version of Miniconda."
+        MINICONDA_URL="https://repo.continuum.io/miniconda"
+        MINICONDA_FILE="Miniconda3-latest-MacOSX-x86_64.sh"
+        curl -L -O "${MINICONDA_URL}/${MINICONDA_FILE}"
+        bash $MINICONDA_FILE -b
+
+        echo "Configuring conda."
+        source ~/miniconda3/bin/activate root
+        conda config --add channels conda-forge
+
+        export CONSTRUCT_ROOT=$PWD
+        bash scripts/build.sh
+        bash scripts/test.sh
```

Comment:
I think it would be cleaner and more consistent to have this in a `build_minifoge_osx.sh` script